src/main/java/de/tum/in/www1/artemis/config/TheiaConfiguration.java

```diff
@@ -0,0 +1,41 @@
+package de.tum.in.www1.artemis.config;
+
+import static de.tum.in.www1.artemis.config.Constants.PROFILE_THEIA;
+
+import java.util.Map;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+
+@Profile(PROFILE_THEIA)
+@Configuration
+@ConfigurationProperties(prefix = "theia")
+public class TheiaConfiguration {
+
+    private Map<String, Map<String, String>> images;
```

Comment:
```suggestion
    private Map<ProgrammingLanguage, Map<String, String>> images;
```
Can you check if this works? If I remember correctly, Spring automatically can parse the enums if the string value is upper or lowercase in `@ConfigurationProperties`. E.g. `java` can still be parsed into `ProgrammingLanguage.JAVA`.

Reply:
`ProgrammingLanguage.JAVA` is coded as `java` anyways, so that wouldn't be a problem :) I tested your suggestion and it works but introduces the challenge of using `ProgrammingLanguage.valueOf()` when getting the map for a language. When I pass a language which is not a language actually, an `java.lang.IllegalArgumentException` is thrown. Do you think the (rather small) advantage of "typing"/using an enum for configuration-validation is worth the extra try-catch in the repository? @b-fein

Reply:
Can you set the type of the request parameter in the resource to `ProgrammingLanguage` as well? Then you would have full type-safety on the server-side without any conversion problems.

Reply:
Done ☺️